sql: add a warning that Range INTERVAL is experimental (#10553)
pingcap/docs: partitioned-table.md

Range COLUMNS partitioning is a variant of Range partitioning. You can use one or more columns as partitioning keys. The data types of partition columns can be integer, string (`CHAR` or `VARCHAR`), `DATE`, and `DATETIME`. Any expressions, such as non-COLUMNS partitioning, are not supported.

> > **Warning:**
> >
> > This is still an experimental feature. It is **NOT** recommended that you use it in the production environment.
> 

Author: ```suggestion
```

### Range INTERVAL partitioning

Range INTERVAL partitioning is an extension of Range partitioning, which allows you to create partitions of a specified interval easily. Starting from v6.3.0, INTERVAL partitioning is introduced in TiDB as syntactic sugar.

> **Warning:**
>

> > This is still an experimental feature. It is **NOT** recommended that you use it in the production environment.

Author: ```suggestion
> This is still an experimental feature. This feature may be changed or removed without prior notice. Usually there are no bugs. In case you find any bugs, please open an issue in the [TiDB repository](https://github.com/pingcap/tidb/issues). 
```

Reviewer: I think the suggestion can be improved further:
```suggestion
> This is an experimental feature. The syntax and implementation may change before GA status.
```
In general I think we should treat released code as production quality, but in this case the syntax is new and an extension of the MySQL dialect, where we want to have some more feedback from community and customers before making the new syntax GA.

Author: OK updated.

> > This is an experimental feature, which might be changed or removed without prior notice. If you find a bug, please open an issue in the [TiDB repository](https://github.com/pingcap/tidb/issues). 

Author: ```suggestion
> This is an experimental feature, which might be changed or removed without prior notice. The syntax and implementation may change before GA status. If you find a bug, please open an issue in the [TiDB repository](https://github.com/pingcap/tidb/issues). 
```

> > This is an experimental feature, which might be changed or removed without prior notice. The syntax and implementation may change before GA status. If you find a bug, please open an issue in the [TiDB repository](https://github.com/pingcap/tidb/issues). 

Author: ```suggestion
> This is an experimental feature, which might be changed or removed without prior notice. The syntax and implementation may change before GA. If you find a bug, please open an issue in the [TiDB repository](https://github.com/pingcap/tidb/issues). 
```
